File: web/src/api/conversation.ts
import apiClient from './client';
import type { ApiResponse } from '@/types/global';

export interface Conversation {
  id: string;
  title: string;
  messages: any[];
  created_at: string;
  updated_at: string;
  is_archived: boolean;
}

export const conversationAPI = {
  // 获取对话列表
  getConversations: (): Promise<ApiResponse<Conversation[]>> => {
    return apiClient.get('/api/conversation');
  },

  // 获取特定对话
  getConversation: (id: string): Promise<ApiResponse<Conversation>> => {
    return apiClient.get(`/api/conversation/${id}`);
  },

  // 创建对话
  createConversation: (data: { title: string }): Promise<ApiResponse<Conversation>> => {
    return apiClient.post('/api/conversation', data);
  },

  // 更新对话
  updateConversation: (id: string, data: Partial<Conversation>): Promise<ApiResponse<Conversation>> => {
    return apiClient.put(`/api/conversation/${id}`, data);
  },

  // 删除对话
  deleteConversation: (id: string): Promise<ApiResponse> => {
    return apiClient.delete(`/api/conversation/${id}`);
  },
};

File: server/router/conversation.go
package router

import (
	"server/api/app"

	"github.com/gin-gonic/gin"
)

// InitConversationRouter 初始化对话相关路由
func InitConversationRouter(privateGroup *gin.RouterGroup, publicGroup *gin.RouterGroup, adminGroup *gin.RouterGroup) {
	// 私有路由 - 用户对话操作
	ConversationRouter := privateGroup.Group("/api/conversation")
	{
		ConversationRouter.GET("", app.GetConversations)          // 获取对话列表
		ConversationRouter.GET("/:id", app.GetConversation)       // 获取特定对话
		ConversationRouter.POST("", app.CreateConversation)       // 创建对话
		ConversationRouter.PUT("/:id", app.UpdateConversation)    // 更新对话
		ConversationRouter.DELETE("/:id", app.DeleteConversation) // 删除对话
	}

	// 目前对话功能没有公共路由和管理员专用路由
	// 如果以后需要可以在这里添加
}
